types/index.go

```diff
@@ -36,7 +37,7 @@ func DefaultIndexKeys(attr string, val *String) []string {
 }
 
 // IntIndex indexs int type.
-func IntIndex(attr string, val *Int32) ([]string, error) {
+func IntIndex(attr string, val *int32) ([]string, error) {
```

Comment:
Just do Index and pass types.Val, so based on the typeid, it would do the right thing.